apps/challenges/views.py

```diff
@@ -340,3 +358,221 @@ def challenge_phase_split_list(request, challenge_pk):
+
+    # Check for evaluation script file in extracted zip folder.
+    if os.path.isfile(evaluation_script_path):
+        challenge_evaluation_script = open(evaluation_script_path, 'rb')
+        challenge_evaluation_script_file = File(challenge_evaluation_script)
+    else:
+        response_data = {
+            'error': 'There is no evaluation script file at the given path in yaml.'
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for test annotation file path in yaml file.
+    yaml_file_data_of_challenge_phase = yaml_file_data['challenge_phases']
+    for data in yaml_file_data_of_challenge_phase:
+        test_annotation_file = data['test_annotation_file']
+        if test_annotation_file:
+            test_annotation_file_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                         unique_folder_name,
+                                                                         extracted_folder_name,
+                                                                         test_annotation_file))
+        else:
+            response_data = {
+                'error': 'Test annotation file path for challenge phase {} does not exist'.format(data['name'])
+                }
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        if not os.path.isfile(test_annotation_file_path):
+            response_data = {
+                'error': 'Test annotation file does not exist at the \
+                         specified path in challenge phase {}'.format(data['name'])
+            }
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for challenge image in yaml file.
+    image = yaml_file_data['image']
+    if image.endswith('.jpg') or image.endswith('.jpeg') or image.endswith('.png'):
+        challenge_image_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                unique_folder_name,
+                                                                extracted_folder_name,
+                                                                image))
+        if os.path.isfile(challenge_image_path):
+            challenge_image = open(challenge_image_path, 'rb')
+            challenge_image_file = File(challenge_image)
+        else:
+            challenge_image_file = None
+    else:
+        challenge_image_file = None
+
+    try:
+        with transaction.atomic():
+            serializer = ZipChallengeSerializer(data=yaml_file_data,
+                                                context={'request': request,
+                                                         'challenge_host_team': challenge_host_team})
+            if serializer.is_valid():
+                challenge = serializer.save()
+                challenge_obj = Challenge.objects.get(pk=challenge.pk)
```

Comment:
Since this block of code is under `transaction.atomic`, it will automatically go into except block on line [571](https://github.com/Cloud-CV/EvalAI/pull/1037/files#diff-ff54996dc8925e4c7dbde340701c7086R571).